src/messaging-component.js

```diff
@@ -0,0 +1,116 @@
+/* globals ppq */
+/* @flow */
+
+import { Tracker } from './tracker-component';
+import { checkIfMobile } from './lib/mobile-check';
+import { loadJavascript } from './lib/load-js';
+import { getCookie, setCookie } from './lib/cookie-utils';
+
+const museSdkUrl = 'https://ppsong.c0d3.com/cdn/assets/modal/sdk.js';
+let userId = '';
+let isRendered = false;
+const sevenDays = 6.048e+8;
+
+const debounce = (f, ms) => {
+    let timeoutId;
+    return (...args : $ReadOnlyArray<mixed>) => {
+        clearTimeout(timeoutId);
+        timeoutId = setTimeout(() => f(...args), ms);
+    };
+};
+
+const showExitModal = ({ cartRecovery }) => {
+    // don't show modal if cartRecovery is not enabled
+    if (!cartRecovery) {
+        return false;
+    }
+    // don't show modal if user is identified
+    const email = getCookie('paypal-cr-user');
+    if (email) {
+        return false;
+    }
+    // don't show modal if user has no cart items
+    const cart = JSON.parse(getCookie('paypal-cr-cart') || '{}');
+    if (!cart.items) {
+        return false;
+    }
+    // don't show modal if user has seen in last 7 days
+    const lastSeen = Boolean(getCookie('paypal-cr-lastseen'));
+    if (lastSeen) {
+        return false;
+    }
+    if (!isRendered) {
+        // $FlowFixMe
+        ppq('updateExperience', {
+            command:    'SHOW_OVERLAY',
+            visitorId:  userId
+        });
+        setCookie('paypal-cr-lastseen', 'true', sevenDays);
+        isRendered = true;
+    }
+    return isRendered;
+};
+
+const init = (...args : $ReadOnlyArray<{ cartRecovery : { userId : string } }>) => {
```

Comment:
For the `Tracker` component y'all went with a constructor pattern: https://github.com/paypal/paypal-muse-components/pull/1 (rather than having an init which modifies globals). For consistency should we do the same here?

```
const messaging = Messaging({
    userID: 'foo'
});
```

Reply:
currently, there's no user actions that are available for messaging component, its just set and go. We could return and empty object to be consistent but I don't think it accomplishes anything. what do you think @bluepnume ?

Reply:
I don't mind if `Messaging` returns null -- mainly thinking about consistency with setting it up and passing in options. Right now:

Tracker -> `Tracker({ ...opts })`
Messaging -> `Messaging.init({ ...opts })`